test/base/RequestBuilder.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package base
+
+import models.requests.{AuthenticatedRequest, VerifiedEmailRequest}
+import models.SignedInUser
+import play.api.mvc.Request
+
+trait RequestBuilder {
+  def decorate[A](sourceRequest: Request[A], user: SignedInUser, email: String = "example@example.com"): VerifiedEmailRequest[A] =
```

Comment:
I find this method's name misleading. It's not clear what it actually does. Could it be renamed to sth like `buuildVerifiedEmailRequest` ?

Reply:
I was thinking along the lines of the method decorates (or wraps) a generic request inside a AuthenticatedRequest and then a VerifiedEmailRequest, but the 'build' concept works just as well so will update :slightly_smiling_face: